mewarpx/mewarpx/sim_control.py

```diff
@@ -0,0 +1,71 @@
+""" Control code used to terminate simulation based on
+a set of user defined criteria
+"""
+
+import logging
+import time
+
+import numpy as np
+import psutil
+
+from pywarpx import _libwarpx
+from mewarpx.mwxrun import mwxrun
+
+# Get module-level logger
+logger = logging.getLogger(__name__)
+
+class SimControl:
+    """
+    Evaluate whether to continue or terminate the simulation based on a
+    set of user defined functions and criteria
+
+    """
+    def __init__(self, criteria, criteria_args, max_loops=None, **kwargs):
```

Comment:
This class should definitely allow for the scenario where `max_loops` is the only argument specified. So I'd suggest giving default values of `None` for `criteria` and `criteria_args` and not setting a default value for max_loops. Then add check in the __init__ function that max_loops >= 1.